src/plugins/controls/public/control_group/component/control_frame_component.tsx

```diff
@@ -42,7 +48,7 @@ export const ControlFrame = ({ customPrepend, enableActions, embeddableId }: Con
   const { overlays } = pluginServices.getHooks();
   const { openConfirm } = overlays.useService();
 
-  const embeddable = useChildEmbeddable({ untilEmbeddableLoaded, embeddableId });
+  const embeddable = useChildEmbeddable({ untilEmbeddableLoaded, embeddableId, embeddableType });
```

Comment:
Need a change in the embeddable type to trigger the `useEffect` of `useChildEmbeddable` so that a re-render happens.